### PR TITLE
chore(types): document store type parameters

### PR DESCRIPTION
**Pull request by gibson042**
Clearing out `git stash`.

Endo analog at https://github.com/endojs/endo/pull/1509

### CODE REVIEW

File: packages/store/src/types.js [hunk 1]
```diff
@@ -184,7 +184,7 @@
  */
 
 /**
- * @template K
+ * @template {Key & object} K
```
Comment (erights) [inline packages/store/src/types.js]:
```suggestion
 * @template {Key} K
```
Not weak. Why `& object`? Seems wrong.

File: packages/store/src/types.js [hunk 2]
```diff
@@ -224,7 +225,8 @@
  */
 
 /**
- * @template K,V
+ * @template {Key & object} K
```
Comment (erights) [inline packages/store/src/types.js]:
```suggestion
 * @template {Key} K
```
Not weak. Why `& object`? Seems wrong.

**Review by erights — APPROVED** [review]
LGTM (notice suggestion)

**Review by turadg — APPROVED** [review]
since `Key` is aliasing `Passable` which is aliasing `any`, I think these are no-ops. still useful as documentation.

fwiw, I usually use `chore(types)` to call that out in the change history. the commit message will already be scoped to the package it's affecting. consider `chore(types): document store type parameters`

**Comment by gibson042** [comment]
These changes actually did uncover some sloppiness in existing code; please review fixes at 7ff2a0c0ca1f9823584d269351f93a1d04c3e7fa.

> since `Key` is aliasing `Passable` which is aliasing `any`, I think these are no-ops. still useful as documentation.

My thoughts exactly, and they will benefit from future refinement of those types.

> fwiw, I usually use `chore(types)` to call that out in the change history. the commit message will already be scoped to the package it's affecting. consider `chore(types): document store type parameters`

Done.

File: packages/inter-protocol/src/vaultFactory/vaultDirector.js [hunk 3]
```diff
@@ -103,6 +105,10 @@ export const prepareVaultDirector = (
 
   // Non-durable map because param managers aren't durable.
   // In the event they're needed they can be reconstructed from contract terms and off-chain data.
+  /**
+   * @template T
+   * @type {MapStore<Brand<AssetKind>, ReturnType<makeVaultParamManager>>}
```
Comment (turadg) [inline packages/inter-protocol/src/vaultFactory/vaultDirector.js]: what's the template for? the value needs a typeof. AssetKind is the default parameter for Brand.
```suggestion
   * @type {MapStore<Brand, ReturnType<typeof makeVaultParamManager>>}
```

File: packages/inter-protocol/test/auction/test-auctionContract.js [hunk 4]
```diff
@@ -102,6 +102,7 @@ const makeAuctionDriver = async (t, customTerms, params = defaultParams) => {
     params,
   );
   const { timerService } = terms;
+  /** @type {MapStore<Brand, { setPrice: (Ratio) => void }>} */
```
Comment (turadg) [inline packages/inter-protocol/test/auction/test-auctionContract.js]: this was copied from `makeManualPriceAuthority` but they both have a bug in that function signature require a param name before the type. without it, the interpretation is `(Ratio: any)`.
```suggestion
  /** @type {MapStore<Brand, { setPrice: (r: Ratio) => void }>} */
```

File: packages/wallet/api/src/lib-wallet.js [hunk 5]
```diff
@@ -178,6 +178,7 @@ export function makeWalletRoot({
   const brandToAutoDepositPurse = makeScalarMapStore('brand');
 
   // Offers that the wallet knows about (the inbox).
+  /** @type {MapStore<Passable, any>} */
```
Comment (turadg) [inline packages/wallet/api/src/lib-wallet.js]: id has to be a string. 
```suggestion
  /** @type {MapStore<string, any>} */
```

**Review by turadg — CHANGES_REQUESTED** [review]

File: packages/inter-protocol/src/vaultFactory/vaultDirector.js (hunk 3 above)
Comment (gibson042) [reply packages/inter-protocol/src/vaultFactory/vaultDirector.js]: Whoops, that's a refactoring typo.

**Review by turadg — APPROVED** [review]
thanks for cleaning up the contract types

**Pull request merged**